liquidation-monitor/src/autotask.spec.js

```diff
@@ -0,0 +1,108 @@
+  shortfallAmount: '1000.00',
+  healthFactor: '0.80',
+};
+const mockBlockHash = '0x1110890564dbd87ca848b7107487ae5a7d28da1b16707bccd3ba37381ae33419';
+const mockTxHash = '0xb85bbcdfd06edf6bcaa3271e49a339cc878daa30ec5f987a43a4d11d925ba751';
+
+const mockFinding = Finding.fromObject({
+  name: 'Placeholder Alert',
+  description: 'Placeholder description',
+  alertId: 'AE-ALERT-ID',
+  type: FindingType.Info,
+  severity: FindingSeverity.Info,
+  protocol: 'Protocol',
+  metadata: mockMetadata,
+});
+
+// grab the existing keys before loading new content from the .env file
+const existingKeys = Object.keys(process.env);
+require('dotenv').config();
+
+// now filter out all of the existing keys from what is currently in the process.env Object
+const newKeys = Object.keys(process.env).filter((key) => existingKeys.indexOf(key) === -1);
+const secrets = {};
+newKeys.forEach((key) => {
+  secrets[key] = process.env[key];
+});
+
+const { handler } = require('./autotask');
+
+function createFortaSentinelEvent(finding, blockHash, txHash) {
+  // Generally findings go from the Bot, to Scan Node, to Sentinel, to Autotasks
+  //  with some metadata being added and removed along the way. This function will mimic
+  // the Sentinel output with only Finding, block and transaction data.
+
+  // Note: Much of the extra data here is superfluous but is left here just in case future bots
+  // want to reference any of the Sentinel data in the Discord output. It also mimics sentinel
+  // output more accurately.
+
+  // On block events, the txHash does not exist
+  if (txHash === undefined || txHash === null) {
+    txHash = '';
+  }
+
+  // populate the matchReasons Array with placeholders
+  const matchReasons = [
+    {
+      type: 'alert-id',
+      value: 'ALERT_ID_PLACEHOLDER',
+    },
+    {
+      type: 'severity',
+      value: 'INFO',
+    },
+  ];
+
+  // populate the sentinel Object with placeholders
+  const sentinel = {
+    id: '8fe3d50b-9b52-44ff-b3fd-a304c66e1e56',
+    name: 'Sentinel Name Placeholder',
+    addresses: [],
+    agents: [],
+    network: 'mainnet',
+    chainId: 1,
+  };
+
+  const autotaskEvent = {
+    relayerARN: undefined,
+    kvstoreARN: undefined,
+    credentials: undefined,
+    tenantId: undefined,
+    secrets,
+    request: {
+      body: {
+        hash: '0xAGENT-HASH', // forta Agent hash
+        alert: {
+          metadata: finding.metadata,
+        },
+        source: {
+          transactionHash: txHash,
+          block: {
+            hash: blockHash,
+          },
+        },
+        matchReasons,
+        sentinel,
+        type: 'FORTA',
+      },
+    },
+  };
+  return autotaskEvent;
+}
+
+describe('check autotask', () => {
+  it('Runs autotask against mock data and posts in Discord (manual-check)', async () => {
+    const autotaskEvent = createFortaSentinelEvent(mockFinding, mockBlockHash, mockTxHash);
+
+    // run the autotask on the events
+    await handler(autotaskEvent);
+  });
```

Comment:
This is a pretty good justification for making the `throw Error` modifications suggested in the Autotask code. If no `.env` file is present, there are no `secrets`, so there is no posting to Discord. However, the logic in the Autotask masks any issue and just returns, so this test succeeds.

We should determine a unified approach to testing all of these Autotasks, as they have very common usages patterns (i.e. Sentinel request in, Discord POST request out), so we will probably be able to develop some reusable mocks that generally cover everything.

Reply:
Added more tests including URL checking and a negative test